day_7/10_speckled_frogs.rb

```diff
@@ -0,0 +1,68 @@
+# OG nursery rhyme
+
+puts '3 speckled frogs sat on a log'
+puts 'eating some most delicious bugs.'
+puts 'One jumped in the pool where its nice and cool,'
+puts 'then there were 2 speckled frogs.'
+puts ''
+puts '2 speckled frogs sat on a log'
+puts 'eating some most delicious bugs.'
+puts 'One jumped in the pool where its nice and cool,'
+puts 'then there was 1 speckled frog.'# removed the s at the end for every version that had one frog for this line. If there's only one it shouldn't be plural.
+puts ''
+puts '1 speckled frog sat on a log'
+puts 'eating some most delicious bugs.'
+puts 'One jumped in the pool where its nice and cool,'
+puts 'then there were no more speckled frogs!'
+puts '-----------------------------------------------'
+# nursery ryhme for 10 frogs
+puts '10 speckled frogs sat on a log'
+puts 'eating some most delicious bugs.'
+puts 'One jumped in the pool where its nice and cool,'
+puts 'then there were 9 speckled frogs.'
+puts ''
+puts '9 speckled frogs sat on a log'
+puts 'eating some most delicious bugs.'
+puts 'One jumped in the pool where its nice and cool,'
+puts 'then there was 8 speckled frogs.'
+puts ''
+puts '8 speckled frog sat on a log'
+puts 'eating some most delicious bugs.'
+puts 'One jumped in the pool where its nice and cool,'
+puts 'then there was 7 speckled frogs.'
+puts '-----------------------------------------------'
+# word version
+puts 'Three speckled frogs sat on a log'
+puts 'eating some most delicious bugs.'
+puts 'One jumped in the pool where its nice and cool,'
+puts 'then there were two speckled frogs.'
+puts ''
+puts 'Two speckled frogs sat on a log'
+puts 'eating some most delicious bugs.'
+puts 'One jumped in the pool where its nice and cool,'
+puts 'then there was one speckled frog.'
+puts ''
+puts 'One speckled frog sat on a log'
+puts 'eating some most delicious bugs.'
+puts 'One jumped in the pool where its nice and cool,'
+puts 'then there were no more speckled frogs!'
+puts '-----------------------------------------------'
+# any number of frogs
+x = 3
+y = 2
+z = 1
+puts "#{x} speckled frogs sat on a log"
+puts 'eating some most delicious bugs.'
+puts 'One jumped in the pool where its nice and cool,'
+puts "then there were #{y} speckled frogs."
+puts ''
+puts "#{y} speckled frogs sat on a log"
+puts 'eating some most delicious bugs.'
+puts 'One jumped in the pool where its nice and cool,'
+puts "then there was #{z} speckled frog."
+puts ''
+puts "#{z} speckled frog sat on a log"
+puts 'eating some most delicious bugs.'
+puts 'One jumped in the pool where its nice and cool,'
+puts 'then there were no more speckled frogs!'
+puts '-----------------------------------------------'
```

Comment:
Try re-doing this exercise using iteration.

Reply:
OK, thanks. I'm going to try and finish up where I'm at in the ruby exercises then give this another go.